compose/compose-file/index.md

```diff
@@ -882,6 +882,81 @@ port (a random host port will be chosen).
      - "127.0.0.1:5000-5010:5000-5010"
      - "6060:6060/udp"
 
+### secrets
+
+After you
+[define or reference secrets in the top-level `secrets` configuration](#secrets-configuration-reference),
+grant access to secrets on a per-service basis using the per-service `secrets`
+configuration. Two different syntax variants are supported.
+
+#### Short syntax
+
+The short syntax variant only specifies the secret name. This grants the
+container access to the secret and mounts it at `/run/secrets/<secret_name>`
+within the container. The source name and destination mountpoint are both set
+to the secret name.
+
+> **Warning**: Due to a bug in Docker 1.13.1, using the short syntax currently
+> mounts the secret with permissions `000`, which means the secret is unreadable
+> if you define it using the short syntax. The workaround is to use the long
+> syntax instead, if you use Docker 1.13.1.
+
+The following example uses the short syntax to grant the `redis` service
+access to the `my_secret` and `my_other_secret` secrets. The value of
+`my_secret` is set to the contents of the file `./my_secret.txt`, and
+`my_other_secret` is defined as an external resource, which means that it has
+already been defined in Docker, outside of this stack file.
```

Comment:
@mstanleyjones It will give you an error:

```
secret not found: external-secret-name
```

> If I understand the declarative model of services correctly, it shouldn't actually be an error to create a service which relies on a secret that doesn't yet exist

It is an error because the client needs to resolve that secret to a secret id. If the secret doesn't exist it can't resolve it, so it can't even create the service.

>  since another stack file could have created it, right? 

Technically yes. What's important is that the user understands they need to create the secret. I don't think the previous wording made that clear.

For external volumes (and networks) we say this:

> If set to `true`, specifies that this volume has been created outside of Compose. `docker-compose up` will not attempt to create it, and will raise an error if it doesn't exist.

Can we use that here as well?